onnxruntime/core/graph/graph.cc

```diff
@@ -2409,7 +2414,28 @@ void Graph::CleanUnusedInitializers() {
   }
 
   std::for_each(erase_list.cbegin(), erase_list.cend(),
-                [this](const std::string& name) { name_to_initial_tensor_.erase(name); });
+                [this](const std::string& name) {
+                  RemoveInitializedTensor(name);
+
+                  // handle edge case where the unused initializer has a matching graph input
```

Comment:
>// handle edge case where the unused initializer has a matching graph input [](start = 18, length = 75)

Do we have test case for this?

Reply:
Some of the ONNX models fail without this change. No direct test case, but given it's pretty rare I didn't think it was necessary to add one. Let me know if you disagree.